### ISSUE TITLE
Official MLton package again?

### ISSUE THREAD

**Issue by achlipala** [Homebrew/homebrew#48150]
MLton is a compiler that is only available as a binary package.  It's really true that today there is no way to build MLton without MLton.  For that reason, a past attempt at a MLton package, in the default Homebrew namespace, wound up in the boneyard, as discussed in #21780.

It's very frustrating for other packages (e.g., Ur/Web, urweb/urweb@a777dd1) not to support easy Homebrew installation, just because they use MLton, which has effectively been ruled out as a package in the default Homebrew namespace.  Is it possible to reconsider, probably with the same kind of treatment that would be applied to other self-hosting compilers like gcc?

The main MLton maintainer @MatthewFluet has what seems like a perfectly reasonable official Homebrew package in MatthewFluet/homebrew-mlton@29327a6.  @eranmeir also developed a MLton package to go with an Ur/Web package in urweb/homebrew-ur@1594cb0f28639fd72722df7f6a85b437f9cd4fe6.

P.S.: Sorry for any terminological slip-ups in this report.  I've never been a Mac user, but I'm the lead developer of Ur/Web, which I'm very eager to have achieve easy-install status in Homebrew.

**Comment by mikemcquaid** [comment]
> MLton is a compiler that is only available as a binary package. It's really true that today there is no way to build MLton without MLton.

It can use a binary version to bootstrap itself but we need it to be built from source. GCC is not self-hosting, technically, because it can be built by Clang. If MLton can be built by another ML compiler and someone submits a formula to build it from source: we'll consider including it. That said: that work needs to be done first.

**Comment by achlipala** [comment]
I don't think that another ML compiler can build MLton anymore.  Is that a kiss of death?  Is it OK for MLton to compile itself in the package?

**Comment by mikemcquaid** [comment]
It's OK for it to compile itself; I was just using that as an example of why this differs from e.g. gcc.

**Comment by MatthewFluet** [comment]
@mikemcquaid Thanks for the feedback.  It isn't really feasible to bootstrap MLton with another SML compiler, so we'll pursue developing a formula that uses the MacOSX binary package for MLton to (re)build from source.

**Comment by DomT4** [comment]
Closing in favour of #48694.
